### PR TITLE
feat: vite flow

### PR DESCRIPTION
**Pull request by Aslemammad**
- vite flow plugin for transforming flow files
- - it has to have include/exclude so we do not just slow the process for any file
- providing transform function for reusability

We need tests for this.

### CODE REVIEW

File: packages/vite-flow/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2021 Callstack
```
Comment (szymonrybczak) [inline packages/vite-flow/LICENSE:3]: Callstack? :D

Comment (Aslemammad) [reply packages/vite-flow/LICENSE:3]: didn't notice that, it was just there after copying the vite swc package in the workspace!

**Comment by Bram-dc** [comment]
Why are you using babel to transform flow types? Would it not be nice to use as little as babel possible and use flow-remove-type instead? Is there more behind that I am missing? Take a look at this minimal example of mine: [vite-plugin-react-native-web](https://github.com/Bram-dc/vite-plugin-react-native-web/blob/master/src/index.ts)

**Comment by szymonrybczak** [comment]
@Bram-dc that's awesome! Thanks for spotting! Would be able to contribute this inside `vxrn`? :pray:

**Comment by Aslemammad** [comment]
@Bram-dc That's amazing, I can send a quick pr right now, but if you want to send one, let me know.

**Comment by Aslemammad** [comment]
I tried, but apparently it does not transform properly? let me know what you think

```
✘ [ERROR] No matching export in "../../packages/vite-native-hmr/dist/esm/index.js" for import "default"

    ../../node_modules/react-native/Libraries/Core/Devtools/loadBundleFromServer.js:13:7:
      13 │ import HMRClient from '../../Utilities/HMRClient';
         ╵        ~~~~~~~~~


 ERROR  Build failed with 1 error:                                                           9:54:21 PM
../../node_modules/react-native/Libraries/Core/Devtools/loadBundleFromServer.js:13:7: ERROR: No matching export in "../../packages/vite-native-hmr/dist/esm/index.js" for import "default"

  ../../node_modules/react-native/Libraries/Core/Devtools/loadBundleFromServer.js:13:7: ERROR: No matching export in "../../packages/vite-native-hmr/dist/esm/index.js" for import "default"
  at failureErrorWithLog (/home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:1651:15)
  at /home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:1059:25
  at runOnEndCallbacks (/home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:1486:45)
  at buildResponseToResult (/home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:1057:7)
  at /home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:1086:16
  at responseCallbacks.<computed> (/home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:704:9)
  at handleIncomingPacket (/home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:764:9)
  at Socket.readFromStdout (/home/aslemammad/oss/vxrn/node_modules/esbuild/lib/main.js:680:7)
  at Socket.emit (node:events:514:28)
  at addChunk (node:internal/streams/readable:545:12)


```

**Comment by Bram-dc** [comment]
@Aslemammad I think you will need to transform the files somehow to fix the imports. Maybe instead of transforming all the files you could fix the issue upstream, or just select specific react-native files to recompile as a hot-fix.

```return (await transformWithEsbuild(code, id, { loader: loader['.js'] })).code```
Since loadBundleFromServer.js end in .js it should get compiled with EsBuild by the plugin. Maybe add some additional options to esbuild to properly transform the imports in loadBundleFromServer.js.

**Comment by Bram-dc** [comment]
Is this using the react-native/metro HMR?
```import HMRClient from '../../Utilities/HMRClient';```

Should vite not be used in this step? No clue how but mixing the two does feel wrong to me.

**Comment by natew** [comment]
We use Vite for the hmr, there’s some legacy code from repack that is now basically unused.

**Pull request merged**